### PR TITLE
postgresql support

### PR DESCRIPTION
**Pull request by Foxite**
I've added PostgreSQL support, to add to the existing MySQL and Sqlite services. Everything was tested and it works. *

I've also slightly simplified `SyncThread.run()` and the `SYNC_TIME` config fields, and I couldn't resist to fix a typo in the config comments.

* If an online player is unwhitelisted from the database (manually or by another server), they will not be kicked from the server on the next sync. This seems to be a problem for MySQL service as well (going by the source code). You can instantly kick all players to who this applies by using `/whitelist reload`, however this is a manual intervention which I'm not sure if it can be done automatically. (You could do `server.getPlayerList().reloadWhitelist()` however, inspecting the source code reveals that the function is empty and does nothing.)

### CODE REVIEW

File: src/main/java/pw/twpi/whitelistsync2/services/PostgreSqlService.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+		} catch (SQLException e) {
+			WhitelistSync2.LOGGER.error("Unexpected exception while copying whitelist to local", e);
+		}
+		return false;
+	}
+
+	@Override
+	public boolean copyDatabaseOppedPlayersToLocal(MinecraftServer server) {
+		if (!Config.SYNC_OP_LIST.get()) {
+			return false;Text is available under the Creative Commons Attribution-ShareAlike License; additional terms may apply. By using this site, you agree to the Terms of Use and Privacy Policy. Wikipedia® is a registered trademark of the Wikimedia Foundation, Inc., a non-profit organization.
```
Comment (PotatoSauceVFX) [inline src/main/java/pw/twpi/whitelistsync2/services/PostgreSqlService.java:195]: Is this text here by accident?

File: build.gradle [hunk 2]
```diff
@@ -60,6 +60,7 @@ configurations {
 }
 
 dependencies {
+    implementation 'org.postgresql:postgresql:42.2.23.jre7'
```
Comment (PotatoSauceVFX) [inline build.gradle:63]: Shouldn't this be shaded like the other libraries below it?

**Review by PotatoSauceVFX — COMMENTED** [review]
Overall, this looks good. Thank you for your contribution. Please see my comments I have added.

File: src/main/java/pw/twpi/whitelistsync2/services/PostgreSqlService.java (hunk 1 above)
Comment (Foxite) [reply src/main/java/pw/twpi/whitelistsync2/services/PostgreSqlService.java]: Yes. I'm not sure how I managed to commit this, as it doesn't compile with it.

File: build.gradle (hunk 2 above)
Comment (Foxite) [reply build.gradle]: Yes. I'll admit I didn't look in this file, I let my IDE add the dependency.

**Review by PotatoSauceVFX — APPROVED** [review]
Looks good. Thank you!

**Comment by PotatoSauceVFX** [comment]
@Foxite After getting around to testing the PostgreSQL support, I ran into a few issues.

1. The whitelist table is not being initially created correctly if it does not exist. The initDatabase is looking for a "whitelist" relation not exist message when it should be looking for "public.whitelist" relation not exists. I don't know anything about this database server so I am not sure how the "public" part factors into this. 
2. I would hope there would be a better way of checking if this table exists that does not involve matching exception errors.
3. The ops table is never created or checked...

Until these issues are corrected, I will be dropping support for Postgresql. The database code has now been moved to [this repo](https://github.com/PotatoSauceVFX/Whitelist-Sync-Lib) so feel free to make a pull request on that repo if you'd still like to see this feature.

**Comment by Foxite** [comment]
Not sure why I declared this finished, when the ops table doesn't work...

I'll see about adding it back properly soon™, if I have time with college and all.

**Comment by PotatoSauceVFX** [comment]
@Foxite No worries! I totally understand.

**Pull request merged**